### PR TITLE
JAMES-2645 Refactor cassandra RRTs by adding a proper DAO class with …

### PR DESCRIPTION
**Pull request by Arsnael**
…basic tests

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[90646779a4df72301c98688a70f444bfe3322fcd] [ERROR] Failures: 
[90646779a4df72301c98688a70f444bfe3322fcd] [ERROR]   CassandraMailboxRecentDAOTest.getRecentMessageUidsInMailboxShouldNotTimeoutWhenOverPagingLimit:123 
[90646779a4df72301c98688a70f444bfe3322fcd] Expected size:<6000> but was:<7000> in:
[90646779a4df72301c98688a70f444bfe3322fcd] <[MessageUid{uid=1},
[90646779a4df72301c98688a70f444bfe3322fcd]     MessageUid{uid=2},
[90646779a4df72301c98688a70f444bfe3322fcd]     MessageUid{uid=3},
[90646779a4df72301c98688a70f444bfe3322fcd]     MessageUid{uid=4},
[...]
```

**Comment by Arsnael** [comment]
test this please

File: server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java [hunk 1]
```diff
@@ -44,135 +30,41 @@
...
-
 public class CassandraRecipientRewriteTable extends AbstractRecipientRewriteTable {
-
-    private final CassandraAsyncExecutor executor;
-    private final CassandraUtils cassandraUtils;
-    private final PreparedStatement insertStatement;
-    private final PreparedStatement deleteStatement;
-    private final PreparedStatement retrieveMappingStatement;
-    private final PreparedStatement retrieveAllMappingsStatement;
+    private final CassandraRecipientRewriteTableDAO cassandraRecipientRewriteTableDAO;
```
Comment (mbaechler) [inline server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java]: as there's a single dao used in this class, maybe you could call it `dao` to make everything shorter ?

File: server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTableDAO.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+public class CassandraRecipientRewriteTableDAO {
+    private final CassandraAsyncExecutor executor;
+    private final CassandraUtils cassandraUtils;
+    private final PreparedStatement insertStatement;
+    private final PreparedStatement deleteStatement;
+    private final PreparedStatement retrieveMappingStatement;
+    private final PreparedStatement retrieveAllMappingsStatement;
+
+    @Inject
+    public CassandraRecipientRewriteTableDAO(Session session, CassandraUtils cassandraUtils) {
```
Comment (mbaechler) [inline server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTableDAO.java]: you probably don't need it to be public

File: server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTableDAO.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                    UserMapping::getSource,
+                    UserMapping::toMapping,
+                    Mappings::union)));
+    }
+
+    private static class UserMapping {
+        private final MappingSource source;
+        private final String mapping;
+
+        public UserMapping(MappingSource source, String mapping) {
```
Comment (mbaechler) [inline server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTableDAO.java]: IMO public methods in a private class doesn't make sense.

File: server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTableDAO.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+    private PreparedStatement prepareInsertStatement(Session session) {
+        return session.prepare(insertInto(TABLE_NAME)
+            .value(USER, bindMarker(USER))
+            .value(DOMAIN, bindMarker(DOMAIN))
+            .value(MAPPING, bindMarker(MAPPING)));
+    }
+
+    CompletableFuture<Void> addMapping(MappingSource source, Mapping mapping) {
+        return executor.executeVoid(insertStatement.bind()
```
Comment (blackheaven) [inline server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTableDAO.java]: Use the Reactor version please

File: server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTableDAO.java [hunk 5]
```diff
@@ -91,33 +92,33 @@ private PreparedStatement prepareInsertStatement(Session session) {
...
+    Mono<Void> removeMapping(MappingSource source, Mapping mapping) {
+        return executor.executeVoidReactor(deleteStatement.bind()
             .setString(USER, source.getFixedUser())
             .setString(DOMAIN, source.getFixedDomain())
             .setString(MAPPING, mapping.asString()));
     }
 
-    CompletableFuture<Optional<Mappings>> retrieveMappings(MappingSource source) {
-        return executor.execute(retrieveMappingStatement.bind()
+    Mono<Optional<Mappings>> retrieveMappings(MappingSource source) {
```
Comment (blackheaven) [inline server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTableDAO.java]: `Mono` should hold the `Optional`ity

**Comment by Arsnael** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**